hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterHeartbeatManager.java

```diff
@@ -160,7 +160,13 @@ public void handleHeartbeat(MembersViewMetadata senderMembersViewMetadata, Strin
 
         try {
             if (!clusterService.isJoined()) {
-                logger.fine("Ignoring heartbeat of sender: " + senderMembersViewMetadata + ", because node is not joined!");
+                if (clusterService.getThisUuid().equals(receiverUuid)) {
+                    logger.fine("Ignoring heartbeat of sender: " + senderMembersViewMetadata + ", because node is not joined!");
```

Comment:
I don't think this will be frequent. At most, this can be logged on every heartbeat (every 5 sec). But this case happens only when node is not joined yet.